pkg/lite-apiserver/cert/cert_mgr.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2020 The SuperEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cert
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/superedge/superedge/pkg/lite-apiserver/config"
+	"k8s.io/klog"
+	"sync"
+	"time"
+)
+
+const (
+	fastReloadDuration = 1 * time.Second
+	reloadDuration     = 30 * time.Minute
+)
+
+type CertManager struct {
+	tlsConfig []config.TLSKeyPair
+
+	fastReload bool
+
+	certMapLock sync.RWMutex
+	certMap     map[string]*tls.Certificate
+}
+
+func NewCertManager(config *config.LiteServerConfig) *CertManager {
+	return &CertManager{
+		tlsConfig: config.TLSConfig,
+		certMap:   make(map[string]*tls.Certificate),
+	}
+}
+
+func (cm *CertManager) Init() error {
+	fastReload := false
+
+	for i := range cm.tlsConfig {
+		cert := cm.tlsConfig[i].CertPath
+		key := cm.tlsConfig[i].KeyPath
+
+		tlsCert, commonName, err := loadCert(cert, key)
+		if err != nil {
+			// load one cert error.
+			fastReload = true
+			continue
+		}
+
+		cm.updateCert(commonName, tlsCert)
+	}
+
+	cm.fastReload = fastReload
+	return nil
+}
+
+func (cm *CertManager) Start() {
+	t := time.NewTimer(cm.getReloadDuration())
+
+	go func() {
+		for {
+			fastReload := false
+
+			select {
+			case <-t.C:
+				for i := range cm.tlsConfig {
```

Comment:
Better to have these in a new function, which will be easier for unit tests as well.

Reply:
add new function handleCertUpdate()